xbmc/pvr/providers/PVRProvider.h

```diff
@@ -0,0 +1,217 @@
+   *
+   * A unique identifier for this provider.
+   * It can be used to find the same provider on this clients backend
+   *
+   * @return The database id of this provider
+   */
+  int GetProviderId() const;
+
+  /*!
+   * @brief Set the database id of this provider
+   * @param iProviderId The new ID.
+   * @return True if the something changed, false otherwise.
+   */
+  bool SetProviderId(int iProviderId);
+
+  /*!
+   * @brief A unique identifier for this provider.
+   *
+   * A unique identifier for this provider.
+   * It can be used to find the same provider on this clients backend
+   *
+   * @return The Unique ID.
+   */
+  int GetUniqueId() const;
+
+  /*!
+   * @return The identifier of the client that supplies this provider.
+   */
+  int GetClientId() const;
+
+  /*!
+   * @return The name of the provider. Can be user provided or the backend name
+   */
+  std::string GetProviderName() const;
+
+  /*!
+   * @brief Set the name of the provider.
+   * @param providerName The new name of the provider.
+   * @return True if the something changed, false otherwise.
+   */
+  bool SetProviderName(const std::string& iProviderName);
+
+  /*!
+   * @brief Checks whether this provider has a type other than
+   *        PVR_PROVIDER_TYPE_UNKNOWN
+   * @return True if this provider has a type other than unknown, false otherwise
+   */
+  bool HasProviderType() const { return m_providerType != PVR_PROVIDER_TYPE_UNKNOWN; }
+
+  /*!
+   * @brief Gets the type of this provider.
+   * @return the type of this provider.
+   */
+  PVR_PROVIDER_TYPE GetProviderType() const;
+
+  /*!
+   * @brief Sets the type of this provider.
+   * @param providerType the new provider type.
+   * @return True if the something changed, false otherwise.
+   */
+  bool SetProviderType(PVR_PROVIDER_TYPE providerType);
+
+  /*!
+   * @brief Get the iconpath for this provider's icon
+   * @return iconpath for this provider's icon
+   */
+  std::string GetIconPath() const;
+
+  /*!
+   * @brief Set the path for this icon
+   * @param strIconPath The new path of the icon.
+   * @return true if the icon path was updated successfully
+   */
+  bool SetIconPath(const std::string& strIconPath);
+
+  /*!
+   * @brief Get this provider's country codes (ISO 3166). Separated by
+   *        PROVIDER_STRING_TOKEN_SEPARATOR.
+   * @return This provider's country codes.
+   */
+  std::string GetCountry() const;
+
+  /*!
+   * @brief Set the country codes for this provider
+   * @param strCountry The new ISO 3166 country codes for this provider. Separated by
+   *                   PROVIDER_STRING_TOKEN_SEPARATOR.
+   * @return true if the country codes were updated successfully
+   */
+  bool SetCountry(const std::string& strCountry);
+
+  /*!
+   * @brief Get this provider's language codes (RFC 5646). Separated by
+   *        PROVIDER_STRING_TOKEN_SEPARATOR.
+   * @return This provider's language codes
+   */
+  std::string GetLanguage() const;
```

Comment:
see my comment above regarding singular/plural and changing the signature

Reply:
Pluralising done